hy/completer.py

```diff
@@ -77,6 +103,20 @@ def complete(self, text, state):
             [matches.append(x) for x in
                 filter(lambda x: x.startswith(text), p)]
 
+        return matches
+
+    def complete(self, text, state):
+
+        matches = []
+
+        if "." in text:
+            try:
+                matches = self.attr_matches(text)
+            except:
+                import traceback
+                traceback.print_exc()
+        else:
+            matches = self.global_matches(text)
         try:
             return matches[state]
```

Comment:
So `matches` can be both a list and a dict, right? Could you add a comment about this?